CMakeLists.txt

```diff
@@ -24,11 +24,7 @@ else()
   cmake_minimum_required(VERSION 3.0)
 endif()
 
-if(USE_CUDA)
-  PROJECT(lightgbm LANGUAGES C CXX CUDA)
-else()
-  PROJECT(lightgbm LANGUAGES C CXX)
-endif()
+PROJECT(lightgbm LANGUAGES C CXX)
```

Comment:
`CMAKE_CUDA_HOST_COMPILER` should be set **before** `PROJECT` (with `CUDA` language) command but `CMAKE_CXX_COMPILER` is available **after** `PROJECT` command execution.

> Once the CUDA language is enabled, the `CMAKE_CUDA_HOST_COMPILER` variable is read-only and changes to it are undefined behavior.
   https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_HOST_COMPILER.html

> And CMake stores path to that default compiler executable in the `CMAKE_C_COMPILER` variable.
   https://stackoverflow.com/a/63944545